tests/xbps.rs

```diff
@@ -0,0 +1,80 @@
+#![cfg(target_os = "linux")]
+
+mod common;
+use common::*;
+
+#[test]
+#[should_panic(expected = "Failed with pattern `^pkgname: wget$`")]
+fn xbps_fail() {
+    test_dsl! {"
+        in -Si fish-shell
+        ou ^pkgname: wget$
+    "}
+}
+
+#[test]
+fn xbps_q() {
```

Comment:
If you feel like doing it, you can also add tests on the special error handling of `-Q` and `-Qe` here :)